app/controllers/users/emails_controller.rb

```diff
@@ -66,7 +66,8 @@ def verify
       if session_email.blank?
         redirect_to add_email_url
       else
-        render :verify, locals: { email: session_email }
+        email_address = EmailAddress.where(user_id: current_user.id).find_with_email(session_email)
+        render :verify, locals: { email: email_address.email, confirmation_token: email_address.confirmation_token }
```

Comment:
This creates a confirmation token variable to use in the `verify.html.erb` below.